src/adapter/src/rbac.rs

```diff
@@ -641,6 +644,7 @@ fn generate_required_ownership(plan: &Plan) -> Vec<ObjectId> {
         Plan::AlterSource(plan) => vec![ObjectId::Item(plan.id)],
         Plan::AlterItemRename(plan) => vec![ObjectId::Item(plan.id)],
         Plan::AlterSecret(plan) => vec![ObjectId::Item(plan.id)],
+        Plan::ValidateConnection(plan) => vec![ObjectId::Item(plan.id)],
```

Comment:
I don't think this is correct, unless you want to restrict validating a connection to the owner of that connection.